cmake/Dependencies.cmake

```diff
@@ -69,6 +69,44 @@ if(BUILD_TEST)
   endif()
 endif()
 
+
+# Benchmark dependencies
+if(BUILD_BENCHMARK)
+  if(NOT DEPENDENCIES_FORCE_DOWNLOAD)
+    # Google Benchmark (https://github.com/google/benchmark.git)
+    find_package(benchmark QUIET)
+  endif()
+
+  if(NOT benchmark_FOUND)
+    message(STATUS "Google Benchmark not found or force download Google Benchmark on. Downloading and building Google Benchmark.")
+    if(CMAKE_CONFIGURATION_TYPES)
+      message(FATAL_ERROR "DownloadProject.cmake doesn't support multi-configuration generators.")
+    endif()
+    set(GOOGLEBENCHMARK_ROOT ${CMAKE_CURRENT_BINARY_DIR}/deps/googlebenchmark CACHE PATH "")
+    if(NOT (CMAKE_CXX_COMPILER_ID STREQUAL "GNU"))
```

Comment:
For HIP on Windows, setting the compiler override to g++ breaks the build.  Can we preserve the old behaviour for HIP on Windows? ie.
if(NOT (CMAKE_CXX_COMPILER_ID STREQUAL "GNU") AND (WIN32 AND CMAKE_CXX_COMPILER MATCHES ".*/hipcc$"))